### DISCUSSION TITLE
Uploading project to GitHub

### DISCUSSION THREAD

**Discussion by Rufida123** [MuhammadSulaiman001/opengl-lab#27] [Q&A] [answered]
I have uploaded my project to GitHub like you said in the first lecture so we can work on the project together as a team, but it seems like my friends are having troubles when cloning my repository like including the libraries and running the code on their laptops even though I have added the libraries to the repository and included them correctly in my project.
so what files should be uploaded so the repository works on all laptops?

**Comment by MuhammadSulaiman001** [comment]
Hi @Rufida123, You might be using full paths for include/lib files.. there is not enough information in your question to give a specific answer.. either provide me with a link to your repo, or at least paste the content of .vcxproj file(s) here in this discussion.

**Reply by Rufida123** [reply]
here is the link to my repo (*******)

Link removed by @MuhammadSulaiman001 :)

**Reply by MuhammadSulaiman001** [reply, marked as answer]
As I've mentioned, you are using full paths for includes and libs, in your .vcxproj file, replace every `C:\Users\rufida\Desktop\OpenGL_repo\OpenGL Code` occurence, with `.\`, for example

replace

```
<IncludePath>C:\Users\rufida\Desktop\OpenGL_repo\OpenGL Code\Libraries;$(IncludePath)</IncludePath>
```

with 

```
<IncludePath>.\Libraries;$(IncludePath)</IncludePath>
```

this way, the path is relative to the .vcxproj file, therefore, your teammates' Visual Studio can find these files and consume them.

Also, I've noticed that you've mentioned another `glad.c` file that is out of your repo!

You should fix that as well, to corrent the path, 

replace

```
<ClCompile Include="..\..\..\OneDrive\Documents\OpenGl\glad.c" />
```

with 

```
<ClCompile Include=".\Libraries\glad.c" />
```

because you have a copy of glad.c at that location.

For more, check `AdditionalIncludeDirectories` and `AdditionalLibraryDirectories` properties in [LabDemos.vcxproj](https://github.com/MuhammadSulaiman001/opengl-lab/blob/master/OpenGLab/LabDemos/LabDemos.vcxproj) as examples, and read [the notes I've added in this repo](https://github.com/MuhammadSulaiman001/opengl-lab/blob/master/docs/Tech%20Tips.md#consuming-libraries).

Note: I've edited your reply to hide the link to your repo (so you keep your homework code hidden😉).

**Reply by Rufida123** [reply]
thank you so much ,I tried your suggestion and everything worked perfectly .
I will consider making my repository private 👍.
